rasa/core/lock_store.py

```diff
@@ -0,0 +1,227 @@
+import asyncio
+import json
+import logging
+import typing
+from typing import Text, Optional, Dict, Union
+
+from async_generator import asynccontextmanager, async_generator, yield_
+
+from rasa.core.lock import TicketLock
+
+if typing.TYPE_CHECKING:
+    pass
+
+logger = logging.getLogger(__name__)
+
+ACCEPTED_LOCK_STORES = ["in_memory", "redis"]
+
+
+class LockError(Exception):
+    pass
+
+
+class LockStore(object):
+    def __init__(self, lifetime: int = 60) -> None:
+        self.lifetime = lifetime
+
+    @staticmethod
+    def find_lock_store(store=None) -> "LockStore":
+        if store is None or store.type is None or store.type == "in_memory":
+            lock_store = InMemoryLockStore()
+        elif store.type == "redis":
+            lock_store = RedisLockStore(host=store.url, **store.kwargs)
+        else:
```

Comment:
should we add support for custom lockStores, means loading them from a module path?

Reply:
at some point yes, not urgent and not in scope of this PR in my opinion. what do you think?

Reply:
It's just one / two lines, isn't it?

Reply:
yeah true - will add it 👍